lib/Optimizer/Transforms/WiresToWiresets.cpp

```diff
@@ -75,18 +75,18 @@ struct AssignWireIndicesPass
   void runOnOperation() override {
     func::FuncOp func = getOperation();
 
+    // Only run on the entrypoint, the expectation is that inlining has been
+    // done already, so there should only be one kernel remaining.
```

Comment:
Seems like this ought to run on pure device kernels too.

Reply:
@schweitzpgi - will you please write an issue for this one (separate from the other dep-analysis issue we're discussing).

Reply:
Created issue.